File: fathersDay/Dsa/Day9 (119. Pascal's Triangle II)/Solution.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<Integer> getRow(int rowIndex) {
        if (rowIndex < 0) {
            return new ArrayList<>(); // Return an empty list for negative rowIndex
        }
        
        // Initialize a list of lists to store the triangle up to rowIndex
        // Each row i will have i+1 elements
        List<List<Integer>> triangle = new ArrayList<>();
        
        for (int i = 0; i <= rowIndex; i++) {
            List<Integer> row = new ArrayList<>();
            for (int j = 0; j <= i; j++) {
                if (j == 0 || j == i) {
                    row.add(1); // First and last elements of each row are 1
                } else {
                    // Each element is the sum of the two elements directly above it
                    row.add(triangle.get(i - 1).get(j - 1) + triangle.get(i - 1).get(j));
                }
            }
            triangle.add(row);
        }
        
        return triangle.get(rowIndex);
    }
}

File: fathersDay/Dsa/Day9 (119. Pascal's Triangle II)/Solution.cpp
#include <vector>

class Solution {
public:
    std::vector<int> getRow(int rowIndex) {
        if (rowIndex < 0) {
            return {}; // Return an empty vector for negative rowIndex
        }
        
        // Initialize a 2D vector to store the triangle up to rowIndex
        // Each row i will have i+1 elements
        std::vector<std::vector<int>> triangle(rowIndex + 1);
        
        for (int i = 0; i <= rowIndex; ++i) {
            triangle[i].resize(i + 1, 1); // Initialize with 1s
        }
        
        // Fill in the values based on Pascal's rule
        for (int i = 2; i <= rowIndex; ++i) {
            for (int j = 1; j < i; ++j) {
                triangle[i][j] = triangle[i - 1][j - 1] + triangle[i - 1][j];
            }
        }
        
        return triangle[rowIndex];
    }
};